src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -133,6 +134,22 @@ public async Task BackgroundUpdateAdvertisingManifestsWhenRequiredAsync()
             return manifestUpdates;
         }
 
+        public IEnumerable<(ManifestId manifestId, ManifestVersion existingVersion, ManifestVersion newVersion)> CalculateManifestRollbacks(string rollbackDefinitionFilePath)
```

Comment:
And i am not sure if we could end up adding or removing manifests

Reply:
I account for that here: https://github.com/dotnet/sdk/pull/18654/files#diff-2a3be4a6207f3297255b699d440fe93813a50baad9c37cf883ffe371853adbe0R144, so we error if the set of manifests defined in the rollback definition doesn't match the set that's currently installed.